R/change_col_type.R

```diff
@@ -0,0 +1,49 @@
+#' @title Change data.table column types
+#'
+#' @description Checks if `check_cols` exist in `dt` and are of type `is_fun`.
+#' If any of `check_cols` are of that  type, then modifies the values using
+#' `as_fun`.
+#'
+#' @param dt \[`data.table()`\] \cr
+#' @param check_cols \[`character()`\] \cr
+#'   name of columns in `dt` to check and modify.
+#' @param is_fun \[`function()`\] \cr
+#'   function like `is.integer()`, `is.character()` etc. to check if each of
+#'   `check_cols` values evaluate to true. Default is `is.integer()`.
+#' @param as_fun \[`function()`\] \cr
+#'   function like `as.integer()`, `as.character()` etc. to change column values
+#'   to. Default is `as.numeric()`.
+#'
+#' @return Modifies `dt` in place and returns `dt` invisibly.
+#'
+#' @examples
+#' dt <- data.table::data.table(year = 1950L, age_start = 0L)
+#' change_col_type(dt, c("year", "year_start", "age_start"))
+#'
+#' @export
+change_col_type <- function(dt,
```

Comment:
Perhaps add a wrapper for this function that will use the arrow `schema` object?

Reply:
That could be nice but would require a bit more work to figure out the mappings between the types. I'm thinking of making an issue with the arrow project to make their functions more flexible. 

The main issue I am having is a variable saved as an integer that the functions can't convert to float/numeric.

```
  # explicitly set the schema for each of the draw dataset columns
  draws_schema:
    process_stage: arrow::string()
    groupings: arrow::string()
    location_id: arrow::int32()
    year_start: arrow::int32()
    sex: arrow::string()
    age_start: arrow::float64()
    age_end: arrow::float64()
    draw: arrow::int32()
    value: arrow::float64()
```